query.py

```diff
@@ -110,10 +110,16 @@ def create_hubmap_metadata_df(
     hubmap_metadata: pd.DataFrame,
     file_types: Iterable[str],
     dataset_class: DataSetClass,
+    assay_filter: Optional[List[str]] = None,
```

Comment:
Please use 3.10+ typehints. Not `Optional`. And please use the most generic type possible. In this case, I think it could be one of `Collection[str]` or `Iterable[str]`.